### DISCUSSION TITLE
Layer2 mode?

### DISCUSSION THREAD

**Discussion by KusakabeShi** [alecbcs/hyprspace#25] [Ideas]
I want to connect all my VPP network by this tool, but VPP's memif have layer2 mode only.

Is there any plan to add l2 mode? using MAC address instead if IP to decide where the packer should go.
Or I have to fork this to modify it by my self.

**Comment by Winterhuman** [comment]
Layer 2 support sounds nice, maybe fork the project, add support for it, and then send a pull request? (Not the developer btw)

**Reply by KusakabeShi** [reply]
I considered again, tap mod and tun mode are not compatible which will break current compatibility.
I think fork this project and create a new one would be a better solution. Now I am doing it!

**Reply by alecbcs** [reply]
If you get a working TAP implementation feel free to submit it as a pull request! I'd love to allow users to pick which ever mode fits their use case best in the an interface's config and be able to switch between TUN/TAP modes.
